planner/core/task.go

```diff
@@ -1322,6 +1323,9 @@ func (p *PhysicalProjection) attach2Task(tasks ...task) task {
 	t = attachPlan2Task(p, t)
 	t.addCost(p.GetCost(t.count()))
 	p.cost = t.cost()
+	if root, ok := tasks[0].(*rootTask); ok && root.isEmpty {
+		t.(*rootTask).isEmpty = true
+	}
```

Comment:
Should we also add this logic to `PhysicalSelection` and `PhysicalLimit`?

Reply:
Of course! But I'm worried about whether we should change the code as less as possible.

Reply:
OK. After rethinking it, `Selection` and `Limit` should be converted into a `TableDual` with the underlying `DataSource` together. So there is no need to update them.